pkg/validate/validate_test.go

```diff
@@ -0,0 +1,185 @@
+package validate
+
+import (
+	"testing"
+
+	"github.com/Azure/acs-engine/pkg/api/osa/vlabs"
+)
+
+/*
+name: openshift
+location: eastus
+properties:
+  openShiftVersion: "$DEPLOY_VERSION"
+  publicHostname: openshift.$RESOURCEGROUP.$DNS_DOMAIN
+  routerProfiles:
+  - name: default
+    publicSubdomain: $RESOURCEGROUP.$DNS_DOMAIN
+  agentPoolProfiles:
+  - name: master
+    role: master
+    count: 3
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: infra
+    role: infra
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: compute
+    role: compute
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  servicePrincipalProfile:
+    clientID: $AZURE_CLIENT_ID
+    secret: $AZURE_CLIENT_SECRET
+*/
+func TestValidate(t *testing.T) {
+
+	tests := map[string]struct {
+		oc            *vlabs.OpenShiftCluster
+		expectedError string
+	}{
+		"invalid version": {
+			oc: &vlabs.OpenShiftCluster{
+				ID:       "test",
+				Location: "eastus",
+				Name:     "testcluster",
+				Plan: &vlabs.ResourcePurchasePlan{
+					Name:      "osa",
+					Product:   "OpenShift",
+					Publisher: "RedHat",
+				},
+				Tags: nil,
+				Type: "",
+				Properties: &vlabs.Properties{
+					OpenShiftVersion: "v3.10",
+					PublicHostname:   "openshift.test.example.com",
+					RouterProfiles: []vlabs.RouterProfile{
+						{
+							Name:            "default",
+							PublicSubdomain: "test.example.com",
+						},
+					},
+					ServicePrincipalProfile: vlabs.ServicePrincipalProfile{
+						ClientID: "test_client_id",
+						Secret:   "test_secret",
+					},
+					AgentPoolProfiles: []vlabs.AgentPoolProfile{
+						{
+							Name:   "master",
+							Role:   "master",
+							Count:  3,
+							VMSize: "Standard_D2s_v3",
+							OSType: "Linux",
+							//VnetSubnetID: "vnet1",
+						},
+						{
+							Name:   "compute",
+							Role:   "compute",
+							Count:  3,
+							VMSize: "Standard_D2s_v3",
+							OSType: "Linux",
+							//VnetSubnetID: "vnet1",
+						},
+						{
+							Name:   "infra",
+							Role:   "infra",
+							Count:  3,
+							VMSize: "Standard_D2s_v3",
+							OSType: "Linux",
+							//VnetSubnetID: "vnet1",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		errs := OpenShiftCluster(test.oc)
+
+		for _, err := range errs {
+
+			//if err.Error != test.expectedError {
+			t.Errorf("Test[%v] failed: expected %v", name, err)
```

Comment:
do a contains on the string here for your error message to ensure you got the right one (will account for wrapping if we end up wrapping in a MSFT helper or something)